internal/cmd/group_flag.go

```diff
@@ -18,9 +22,29 @@ func addGroupFlag(cmd *cobra.Command) {
 }
 
 var fromDBFlag string
+var timestampFlag string
 
 func addFromDBFlag(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&fromDBFlag, "from-db", "", "Creates the new database based on an existing one")
 	cmd.Flags().MarkHidden("from-db")
 	cmd.RegisterFlagCompletionFunc("from-db", dbNameArg)
+
+	cmd.Flags().StringVar(&timestampFlag, "timestamp", "", "When used with --from-db option, new database will represent state of its origin at given point in time")
+	cmd.Flags().MarkHidden("timestamp")
+}
+
+func parseTimestampFlag() (*time.Time, error) {
+	var timestamp *time.Time
+	if fromDBFlag != "" {
+		if timestampFlag != "" {
+			ts, err := time.Parse("2006-01-02T15:04:05", timestampFlag)
+			if err != nil {
+				return nil, fmt.Errorf("provided timestamp was not in 'yyyy-MM-ddThh:mm::ss' format")
+			}
+			timestamp = &ts
+		}
+	} else if timestampFlag != "" {
+		return nil, fmt.Errorf("--timestamp cannot be used without specifying --from-db option")
+	}
+	return timestamp, nil
 }
```

Comment:
⛏️: I suggest the following refactoring:
```suggestion
func parseTimestampFlag() (*time.Time, error) {
	if timestampFlag == "" {
		return nil, nil
	}
	if fromDBFlag == "" {
		return nil, fmt.Errorf("--timestamp cannot be used without specifying --from-db")
	}

	ts, err := time.Parse("2006-01-02T15:04:05", timestampFlag)
	if err != nil {
		return nil, fmt.Errorf("provided timestamp was not in 'yyyy-MM-ddThh:mm::ss' format")
	}
	return ts, nil
}
```
With it, you'll explicitly handle the cases you want to handle while avoiding nesting conditional logic.

Reply:
@Horusiath as @haaawk they are not functionally equivalent. Yours is missing the case where both flags are empty.
You should also address my other suggestion, which is preventing errors from being displayed to the user.